core/mosaic-webpack-fallback-plugin/lib/source-directories.js

```diff
@@ -0,0 +1,17 @@
+const { getMosaicConfig } = require('@tilework/mosaic-dev-utils/mosaic-config');
+/**
+ * Get sourceDirectories field from packageJson.
+ *
+ * Fallback to 'src' and 'pub' if no field is found.
+ * @param {string} cwd
+ * @returns {string[]}
+ */
+const prepareSourcesDirectories = (cwd = process.cwd()) => {
+    const { sourceDirectories: projectSourceDirectories = [] } = getMosaicConfig(cwd);
+
+    return ['src', 'pub'].concat(projectSourceDirectories);
+};
```

Comment:
Such an implementation will not be capsule-like - one package will be able to add a fallback directory to another package. We want them to only add fallback directories to themselves. I propose to:

1. When preparing source directories, instead of relative paths store actual resolved paths of the directories. Use `path.resolve` for that.
2. When checking whether fallback is necessary, you will need to `sourceDirectories.some(dir => thepath.startsWith(dir))`

Reply:
I see, will do!